locator/tablets.cc

```diff
@@ -132,6 +132,12 @@ future<> tablet_metadata::clear_gently() {
     co_return;
 }
 
+void tablet_metadata::merge(tablet_metadata&& other) {
+    for (auto&& [id, map] : other._tablets) {
```

Comment:
I think this doesn't work with deletions. If there is a hint, but element is absent in other, we should delete.